mitmproxy/filt.py

```diff
@@ -193,12 +193,22 @@ class FBod(_Rex):
     help = "Body"
 
     def __call__(self, f):
-        if f.request and f.request.content:
-            if self.re.search(f.request.get_decoded_content()):
-                return True
-        if f.response and f.response.content:
-            if self.re.search(f.response.get_decoded_content()):
-                return True
+
+        # HTTPFlow
+        if hasattr(f, 'request'):
```

Comment:
We can also support decorator chaining, e.g.
```python
@http
@tcp
def __call__(self, f):
    # this filter only works for http and tcp flows
```
This is a bit funky to implement, but certainly doable - I'll leave the challenge for you. :blush: